cl/corpus_importer/management/commands/troller_bk.py

```diff
@@ -31,12 +37,77 @@
     hash_item,
     is_cached,
 )
-from cl.search.models import Court
+from cl.search.models import Court, Docket, DocketEntry, RECAPDocument
 from cl.search.tasks import add_items_to_solr
 
 FILES_BUFFER_THRESHOLD = 3
 
 
+def get_docket_entries_to_add(
+    court_id: str, des_to_add_existing_docket: list[tuple[int, dict[str, Any]]]
+) -> list[DocketEntry]:
+    """Make and return a list of the DocketEntry objects to save in bulk.
+
+    :param court_id: The court the docket entries belong to.
+    :param des_to_add_existing_docket: A list of tuples containing the docket
+    pk the entry belongs to, the docket entry dict.
+    :return: A list of DocketEntry objects.
+    """
+
+    docket_entries_to_add_bulk = []
+    for de_add in des_to_add_existing_docket:
+        d_pk = de_add[0]
+        docket_entry = de_add[1]
+        calculate_recap_sequence_numbers([docket_entry], court_id)
+        date_filed, time_filed = localize_date_and_time(
+            court_id, docket_entry["date_filed"]
+        )
+        de_to_add = DocketEntry(
+            docket_id=d_pk,
+            entry_number=docket_entry["document_number"],
+            description=docket_entry["description"],
+            pacer_sequence_number=docket_entry["pacer_seq_no"],
+            recap_sequence_number=docket_entry["recap_sequence_number"],
+            time_filed=time_filed,
+            date_filed=date_filed,
+        )
+        docket_entries_to_add_bulk.append(de_to_add)
+    return docket_entries_to_add_bulk
+
+
+def get_rds_to_add(
+    des_bulk_created: list[DocketEntry],
+    des_to_add_existing_docket: list[tuple[int, dict[str, Any]]],
+) -> list[RECAPDocument]:
+    """Make and return a list of the RECAPDocument objects to save in bulk.
+
+    :param des_bulk_created: The list of DocketEntry objects saved in a
+    previous step.
+    :param des_to_add_existing_docket: A list of tuples containing the docket
+    pk the entry belongs to, the docket entry dict.
+    :return: A list of RECAPDocument objects.
+    """
+
+    rds_to_create_bulk = []
+    index_de = 0
+    for d_entry in des_to_add_existing_docket:
+        de_pk = des_bulk_created[index_de].pk
```

Comment:
The index stuff here feels pretty wonky. I'm not sure a better way to do it, but it might be worth a second look.

Reply:
Yeah, I checked this and instead of using an `index`, now using `zip()` to iterate in both lists at the same time, so no need to use the `index`